contracts/payloads/IGP85/PayloadIGP85.sol

```diff
@@ -0,0 +1,548 @@
+    function action3() internal isActionSkippable(3) {
+        {
+            address ezETH_ETH_DEX = getDexAddress(21);
+            // ezETH-ETH DEX
+            {
+                // ezETH-ETH Dex
+                Dex memory DEX_ezETH_ETH = Dex({
+                    dex: ezETH_ETH_DEX,
+                    tokenA: ezETH_ADDRESS,
+                    tokenB: ETH_ADDRESS,
+                    smartCollateral: true,
+                    smartDebt: false,
+                    baseWithdrawalLimitInUSD: 7_500_000, // $7.5M
+                    baseBorrowLimitInUSD: 0, // $0
+                    maxBorrowLimitInUSD: 0 // $0
+                });
+                setDexLimits(DEX_ezETH_ETH); // Smart Collateral
+
+                DEX_FACTORY.setDexAuth(ezETH_ETH_DEX, TEAM_MULTISIG, false);
+            }
+        }
+
+        {
+            address ezETH__wstETH_VAULT = getVaultAddress(103);
+
+            // [TYPE 1] ezETH<>wstETH | normal collateral & normal debt
+            Vault memory VAULT_ezETH_wstETH = Vault({
+                vault: ezETH__wstETH_VAULT,
+                vaultType: TYPE.TYPE_1,
+                supplyToken: ezETH_ADDRESS,
+                borrowToken: wstETH_ADDRESS,
+                baseWithdrawalLimitInUSD: 10_000_000, // $10M
+                baseBorrowLimitInUSD: 10_000_000, // $10M
+                maxBorrowLimitInUSD: 20_000_000 // $20M
+            });
+
+            setVaultLimits(VAULT_ezETH_wstETH); // TYPE_1 => 103
+
+            VAULT_FACTORY.setVaultAuth(
+                ezETH__wstETH_VAULT,
+                TEAM_MULTISIG,
+                false
+            );
+        }
+
+        {
+            address ezETH_ETH__wstETH_VAULT = getVaultAddress(104);
+
+            // [TYPE 2] ezETH-ETH<>wstETH | smart collateral & normal debt
+            Vault memory VAULT_ezETH_ETH_wstETH = Vault({
+                vault: ezETH_ETH__wstETH_VAULT,
+                vaultType: TYPE.TYPE_2,
+                supplyToken: address(0),
+                borrowToken: wstETH_ADDRESS,
+                baseWithdrawalLimitInUSD: 0,
+                baseBorrowLimitInUSD: 7_500_000, // $7.5M
+                maxBorrowLimitInUSD: 15_000_000 // $15M
+            });
+
+            setVaultLimits(VAULT_ezETH_ETH_wstETH); // TYPE_2 => 104
+
+            VAULT_FACTORY.setVaultAuth(
+                ezETH_ETH__wstETH_VAULT,
+                TEAM_MULTISIG,
+                false
+            );
+        }
+    }
+
+    // @notice Action 4: Update rewards for fUSDC, fUSDT
+    function action4() internal isActionSkippable(4) {
+        address[] memory protocols = new address[](2);
+        address[] memory tokens = new address[](2);
+        uint256[] memory amounts = new uint256[](2);
+
+        address REWARDS_ADDRESS = address(
+            //0x
+        );
+
+        {
+            /// fUSDC
+            IFTokenAdmin(F_USDC).updateRewards(REWARDS_ADDRESS);
+
+            uint256 allowance = IERC20(USDC_ADDRESS).allowance(
```

Comment:
whole allowance logic can be removed, only need to call updateRewards for both. No need to call .start() either, we will use a new contract that can be configured by MS